platform/asterix/boot/src/drivers/pmic.c

```diff
@@ -91,94 +141,83 @@ int pmic_init(void) {
-  if (ret != 0) {
-    return ret;
-  }
-
-  ret = prv_pmic_write(CHARGER_BASE, BCHGVTERMR, BCHGVTERMREDUCED_4V00);
-  if (ret != 0) {
-    return ret;
-  }
-
-  ret = prv_pmic_write(CHARGER_BASE, BCHGISETMSB, 16);
-  if (ret != 0) {
-    return ret;
-  }
-
-  ret = prv_pmic_write(CHARGER_BASE, BCHGISETDISCHARGEMSB, 16);
-  if (ret != 0) {
-    return ret;
-  }
-
-  ret = prv_pmic_write(CHARGER_BASE, BCHGENABLESET, ENABLECHARGING_ENABLECHG);
-  if (ret != 0) {
-    return ret;
-  }
-
-  // LDO2 as LDO @ 1.8V (powers the QSPI flash)
-  ret = prv_pmic_write(LDSW_BASE, LDSW2LDOSEL, LDSW2LDOSEL_LDO);
-  if (ret != 0) {
-    return ret;
-  }
-
-  ret = prv_pmic_write(LDSW_BASE, LDSW2VOUTSEL, LDSW2VOUTSEL_1V8);
-  if (ret != 0) {
-    return ret;
-  }
-
-  ret = prv_pmic_write(LDSW_BASE, TASKLDSW2SET, 0x01U);
-  if (ret != 0) {
-    return ret;
+  const struct pmic_reg regs[] = {
+    // Turn off any watchdog / boot timer right away.
+    { TIMER_BASE, TIMERCLR, TIMERCLR_TASKTIMERDIS },
+    { ERRLOG_BASE, SCRATCH0, 0x00 /* contains boot timer bit */ },
+    
+    // Make sure right away that we can reset the device if needed.
+    { SHIP_BASE, LPRESETCFG, 0 },
+    
+    // Set up the BUCK1 regulator for manual control to 1.8V, automatic
+    // PWM/hysteresis control.
+    { BUCK_BASE, BUCK1ENASET, 1 },
+    { BUCK_BASE, BUCK2ENASET, 1 },
+    { BUCK_BASE, BUCK1PWMCLR, 1 },
+    { BUCK_BASE, BUCK2PWMCLR, 1 },
+    { BUCK_BASE, BUCK1NORMVOUT, BUCKVOUT_1V8 },
+    { BUCK_BASE, BUCK1RETVOUT, BUCKVOUT_1V8 },
+    { BUCK_BASE, BUCK2NORMVOUT, BUCKVOUT_3V0 },
+    { BUCK_BASE, BUCK2RETVOUT, BUCKVOUT_3V0 },
+    { BUCK_BASE, BUCKENCTRL, 0 },
+    { BUCK_BASE, BUCKVRETCTRL, 0 },
+    { BUCK_BASE, BUCKPWMCTRL, 0 },
+    { BUCK_BASE, BUCKSWCTRLSET, BUCKSWCTRLSET_BUCK1SWCTRLSET | BUCKSWCTRLSET_BUCK2SWCTRLSET /* use registers rather than resistor settings */ },
+    { BUCK_BASE, BUCKCTRL0, 0 },
+
+    // Configure charger (TODO: values are board/battery dependent)
+    // - Thermistor: 10K NTC
+    // - Termination voltage: 4.2V
+    // - Reduced termination voltage (for warm region): 4.00V
+    // - Charge current limit of 152 mA (approximately 1C for most reasonable wearable batteries)
+    // - Discharge current limit of 200 mA (increase current measurement accuracy)
+    // - Release charger from error state if applicable (but do not clear
+    //   safety timers) -- this doesn't happen in a loop because after we
+    //   fail to boot three times, we will sit at sadwatch until a button is
+    //   pressed
+    // - Enable charging
+    { VBUSIN_BASE, VBUSINILIMSTARTUP, VBUSLIM_500MA }, // should be default, but 'reset value from OTP, value listed in this table may not be correct'
+    { CHARGER_BASE, BCHGENABLECLR, ENABLECHARGING_DISABLECHG },
+    { ADC_BASE, ADCNTCRSEL, ADCNTCRSEL_10K },
+    { CHARGER_BASE, BCHGVTERM, BCHGVTERMNORM_4V20 },
+    { CHARGER_BASE, BCHGVTERMR, BCHGVTERMREDUCED_4V00 },
+    { CHARGER_BASE, BCHGISETMSB, 38 },
```

Comment:
0.5C may still be a safer limit here, considering discharge can be adjusted independently

Reply:
Really, 1C will be fine.  (Actually, 1C will be too slow for most users and we will want to turn it up even further in firmware.)  But 1C is definitely safe for the bootloader to do.